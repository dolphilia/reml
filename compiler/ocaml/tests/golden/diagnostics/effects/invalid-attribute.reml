@requires_capability(true)
fn demo() -> i64 {
  0
}
