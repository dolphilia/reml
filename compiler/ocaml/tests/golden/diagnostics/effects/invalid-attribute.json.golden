{
  "diagnostics": [
    {
      "audit": {
        "metadata": {
          "effect.stage.required": "at_least:stable",
          "effect.stage.actual": "stable",
          "effect.capability": "runtime",
          "effect.attribute.invalid": "stage",
          "effect.attribute.location": "demo",
          "effect.attribute.reason": "unsupported_stage_value",
          "stage_trace": [
            {
              "source": "cli_option",
              "note": "not provided"
            },
            {
              "source": "env_var",
              "note": "REMLC_EFFECT_STAGE not set"
            },
            {
              "source": "typer",
              "stage": "stable"
            }
          ],
          "schema.version": "1.1",
          "audit.timestamp": "1970-01-01T00:00:00Z",
          "audit.policy.version": "phase2.5.audit.v1",
          "audit.channel": "cli",
          "audit.build_id": "19700101T000000Z",
          "audit.sequence": 0,
          "audit.source.workspace": ".",
          "cli.audit_id": "cli/19700101T000000Z#0",
          "cli.change_set": {
            "policy": "phase2.5.audit.v1",
            "origin": "cli",
            "source": {
              "build_id": "19700101T000000Z",
              "workspace": ".",
              "sequence": 0
            },
            "items": [
              {
                "kind": "cli-command",
                "command": "effects-invalid-attribute"
              },
              {
                "kind": "input",
                "path": "<入力>"
              }
            ]
          },
          "event.domain": "type",
          "event.kind": "effects.syntax.invalid_attribute"
        },
        "audit_id": "cli/19700101T000000Z#0",
        "change_set": {
          "policy": "phase2.5.audit.v1",
          "origin": "cli",
          "source": {
            "build_id": "19700101T000000Z",
            "workspace": ".",
            "sequence": 0
          },
          "items": [
            {
              "kind": "cli-command",
              "command": "effects-invalid-attribute"
            },
            {
              "kind": "input",
              "path": "<入力>"
            }
          ]
        }
      },
      "extensions": {
        "effects": {
          "diagnostic_payload": {
            "invalid_attributes": [
              {
                "attribute": "@requires_capability(true)",
                "attribute_name": "requires_capability",
                "reason": "unsupported_stage_value",
                "key": "stage",
                "provided": true,
                "provided_display": "true"
              }
            ],
            "residual_leaks": []
          },
          "invalid_attributes": [
            {
              "attribute": "@requires_capability(true)",
              "attribute_name": "requires_capability",
              "reason": "unsupported_stage_value",
              "key": "stage",
              "provided": true,
              "provided_display": "true"
            }
          ],
          "stage": {
            "required": "at_least:stable",
            "actual": "stable"
          },
          "provided": true,
          "expected": "stage は文字列または StageId",
          "attribute": "@requires_capability(true)",
          "stage_trace": [
            {
              "source": "cli_option",
              "note": "not provided"
            },
            {
              "source": "env_var",
              "note": "REMLC_EFFECT_STAGE not set"
            },
            {
              "source": "typer",
              "stage": "stable"
            }
          ],
          "key": "stage",
          "provided_display": "true"
        },
        "effect.stage.required": "at_least:stable",
        "effect.stage.actual": "stable",
        "effect.stage.capability": "runtime",
        "effect.stage.subject": "demo",
        "effect.stage_trace": [
          {
            "source": "cli_option",
            "note": "not provided"
          },
          {
            "source": "env_var",
            "note": "REMLC_EFFECT_STAGE not set"
          },
          {
            "source": "typer",
            "stage": "stable"
          }
        ],
        "effect.invalid_attributes": [
          {
            "attribute": "@requires_capability(true)",
            "attribute_name": "requires_capability",
            "reason": "unsupported_stage_value",
            "key": "stage",
            "provided": true,
            "provided_display": "true"
          }
        ]
      },
      "notes": [
        "stage は文字列または StageId",
        "指定された値: true"
      ],
      "domain": "type",
      "code": "effects.syntax.invalid_attribute",
      "severity": "error",
      "message": "効果属性 @requires_capability(true) は無効です",
      "location": {
        "file": "<入力>",
        "line": 1,
        "column": 22,
        "endLine": 1,
        "endColumn": 26
      },
      "timestamp": "1970-01-01T00:00:00Z",
      "audit_metadata": {
        "effect.stage.required": "at_least:stable",
        "effect.stage.actual": "stable",
        "effect.capability": "runtime",
        "effect.attribute.invalid": "stage",
        "effect.attribute.location": "demo",
        "effect.attribute.reason": "unsupported_stage_value",
        "stage_trace": [
          {
            "source": "cli_option",
            "note": "not provided"
          },
          {
            "source": "env_var",
            "note": "REMLC_EFFECT_STAGE not set"
          },
          {
            "source": "typer",
            "stage": "stable"
          }
        ],
        "schema.version": "1.1",
        "audit.timestamp": "1970-01-01T00:00:00Z",
        "audit.policy.version": "phase2.5.audit.v1",
        "audit.channel": "cli",
        "audit.build_id": "19700101T000000Z",
        "audit.sequence": 0,
        "audit.source.workspace": ".",
        "cli.audit_id": "cli/19700101T000000Z#0",
        "cli.change_set": {
          "policy": "phase2.5.audit.v1",
          "origin": "cli",
          "source": {
            "build_id": "19700101T000000Z",
            "workspace": ".",
            "sequence": 0
          },
          "items": [
            {
              "kind": "cli-command",
              "command": "effects-invalid-attribute"
            },
            {
              "kind": "input",
              "path": "<入力>"
            }
          ]
        },
        "event.domain": "type",
        "event.kind": "effects.syntax.invalid_attribute"
      }
    }
  ]
}
